core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -216,7 +218,7 @@ public String getUpdateType()
     {
         return target.map(target -> {
             QualifiedObjectName name = target.getName();
-            return new Output(name.getCatalogName(), name.getSchemaName(), name.getObjectName());
+            return new Output(name.getCatalogName(), name.getSchemaName(), name.getObjectName(), targetColumns);
```

Comment:
agree with @kokosing that it'd be better to keep this closer to setUpdateType, as that's what we're trying to follow here. could you share what difference we'll see in terms of error message - that prevents this change?

If possible, would it be better to make targetColumns part of target and update target object?

Reply:
The only difference we would see in `CREATE TABLE IF NOT EXISTS` - where `setUpdate` might not be triggered if the table is already present. One workaround is to invoke `setUpdate` at two places.